core/trino-main/src/main/java/io/trino/operator/scalar/JsonStringArrayExtractScalar.java

```diff
@@ -0,0 +1,161 @@
+                .signature(Signature.builder()
+                        .argumentType(new TypeSignature("varchar", typeVariable("N")))
+                        .longVariable("N")
+                        .argumentType(new TypeSignature(JsonPathType.NAME))
+                        .returnType(arrayType(VARCHAR.getTypeSignature()))
+                        .build())
+                .nullable()
+                .hidden()
+                .noDescription()
+                .build());
+    }
+
+    @Override
+    protected SpecializedSqlScalarFunction specialize(BoundSignature boundSignature)
+    {
+        return new ChoicesSpecializedSqlScalarFunction(
+                boundSignature,
+                NULLABLE_RETURN,
+                ImmutableList.of(NEVER_NULL, NEVER_NULL),
+                METHOD_HANDLE);
+    }
+
+    @UsedByGeneratedCode
+    public static Block extract(Slice json, JsonPath jsonPath)
+    {
+        ArrayType arrayType = new ArrayType(VARCHAR);
+        ArrayBlockBuilderAppender arrayAppender = new ArrayBlockBuilderAppender();
+        try (JsonParser jsonParser = createJsonParser(JSON_FACTORY, json)) {
+            jsonParser.nextToken();
+            if (jsonParser.getCurrentToken() == JsonToken.VALUE_NULL) {
+                return null;
+            }
+            BlockBuilder blockBuilder = arrayType.createBlockBuilder(null, 1);
+            arrayAppender.append(jsonParser, jsonPath, blockBuilder);
+            Block block = blockBuilder.build();
+            return arrayType.getObject(block, 0);
+        }
+        catch (TrinoException | JsonCastException e) {
+            throw new TrinoException(INVALID_CAST_ARGUMENT, format("Cannot cast to %s. %s\n%s", arrayType, e.getMessage(), truncateIfNecessaryForErrorMessage(json)), e);
+        }
+        catch (Exception e) {
+            throw new TrinoException(INVALID_CAST_ARGUMENT, format("Cannot cast to %s.\n%s", arrayType, truncateIfNecessaryForErrorMessage(json)), e);
+        }
+    }
+
+    public record ArrayBlockBuilderAppender()
+    {
+        public void append(JsonParser parser, JsonPath path, BlockBuilder blockBuilder)
+                throws IOException
+        {
+            if (parser.getCurrentToken() == JsonToken.VALUE_NULL) {
+                blockBuilder.appendNull();
+                return;
+            }
+
+            if (parser.getCurrentToken() != START_ARRAY) {
+                throw new JsonCastException(format("Expected a json array, but got %s", parser.getText()));
+            }
+            ((ArrayBlockBuilder) blockBuilder).buildEntry(elementBuilder -> {
+                while (parser.nextToken() != END_ARRAY) {
+                    if (parser.getCurrentToken() == JsonToken.VALUE_NULL) {
+                        append(null, elementBuilder);
+                        continue;
+                    }
+                    JsonParser jsonParser = parser.readValueAsTree().traverse();
```

Comment:
Right now, extracting from jsonParser is specific to this class only. This could be next step if we will make more similar class (for rows/maps for example) that avoid serialization/deserialization between json function calls.